packages/vaex-core/vaex/arrow/dataset.py

```diff
@@ -151,19 +185,68 @@ def chunk_iterator(self, columns, chunk_size=None, reverse=False, start=0, end=N
 
 
 
+# just what we need to support the scanning of the Hiva layout
+class LegacyWrapper(pa.filesystem.FileSystem):
+    def __init__(self, fs):
+        self.fs = fs
+
+    def isdir(self, path):
+        return self.fs.get_file_info([path])[0].type == pa.fs.FileType.Directory
+
+    def isfile(self, path):
+        return self.fs.get_file_info([path])[0].type == pa.fs.FileType.File
+
+    def _isfilestore(self):
+        return False
+
+    def open(self, path, mode='rb'):
+        path = vaex.file.stringyfy(path)
+        assert mode == 'rb'
+        return self.fs.open_input_file(path)
+
+    def ls(self, path, detail=False):
+        path = vaex.file.stringyfy(path)
+        return self.fs.ls(path, detail=detail)
+
+    def walk(self, path):
+        path = vaex.file.stringyfy(path)
+        infos = self.fs.get_file_info(pa.fs.FileSelector(path))
+        directories = [k.path[len(path):] for k in infos if k.type == pa.fs.FileType.Directory]
+        files = [k.path[len(path):] for k in infos if k.type == pa.fs.FileType.File]
+        yield path, directories, files
+
+
 class DatasetParquet(DatasetArrowBase):
-    def __init__(self, path, fs_options, max_rows_read=1024**2*10):
+    def __init__(self, path, fs_options, max_rows_read=1024**2*10, partitioning=None, kwargs=None):
         self.path = path
         self.fs_options = fs_options
+        self.partitioning = partitioning
+        self.kwargs = kwargs or {}
         super().__init__(max_rows_read=max_rows_read)
 
     def _create_dataset(self):
-        file_system, path = vaex.file.parse(self.path, self.fs_options, for_arrow=True)
-        self._arrow_ds = pyarrow.dataset.dataset(path, filesystem=file_system)
+        file_system, source = vaex.file.parse(self.path, self.fs_options, for_arrow=True)
+        # we need to mix both arrow.parquet and arrow.dataset since the latter does not expose
+        # the partitions
+        if file_system is not None:
+            file_system_legacy = LegacyWrapper(file_system)
+        else:
+            file_system_legacy = LegacyWrapper(pa.fs.LocalFileSystem())
+        pds = pq.ParquetDataset(source,
```

Comment:
Is there a specific reason (feature) for which you need to use the pq.ParquetDataset instead of `ds.dataset(..)` ? 

(since `pyarrow.dataset` is meant as a replacement, that's interesting to know)

Reply:
Yes, I didn't open an issue on this yet for pyarrow. The partition information is not exposed AFAIK, so I cannot access the equivalent of `pds.partitions`.

Reply:
It is available, but a bit hidden .. 
Each fragment has a `partition_expression`. Now, because the public API of those expression objects is limited, you can't really get the needed information out of this object. 
For that reason (and because dask also needs it), we added a private helper function for now:

```
In [1]: import pyarrow.dataset as ds

In [3]: dataset = ds.dataset("notebooks-arrow/nyc-taxi-data/dask-partitioned/", partitioning="hive")

In [4]: fragment = list(dataset.get_fragments())[0]

In [5]: fragment.partition_expression
Out[5]: <pyarrow.dataset.Expression ((year == 2016:int32) and (month == 10:int32))>

In [6]: ds._get_partition_keys(fragment.partition_expression)
Out[6]: {'month': 10, 'year': 2016}
```

It's "private-ish", mainly because we probably should come up with a better interface for this, but since advanced users of pyarrow.dataset like dask or vaex need it on the short term, it was added (and we can commit to keep it working)

Reply:
(I know we need to improve this API .. any feedback on the information you need from the dataset is very welcome! We don't want to copy exactly the ParquetDataset API with its `pieces`, but at least we should somehow enable similar functionality)

Reply:
That is awesome, I explored all those options but never got anywhere. Thanks a lot, Joris! I was ' ' this close to reaching out, but there is not low barrier way or reaching out, I find the mailing list a bit high barrier for some reason.

I kinda liked the ParquetDataset, with the dictionary, because now I have to assemble it myself.